person.rb

```diff
@@ -19,3 +30,26 @@ def of_age?
     @age >= 18
   end
 end
+
+class Decorator < Nameable
+  def initialize(nameable)
+    super()
+    @nameable = nameable
+  end
+
+  def correct_name
+    @nameable.correct_name
+  end
+end
+
+class CapitalizeDecorator < Decorator
+  def correct_name
+    @nameable.correct_name.capitalize
+  end
+end
+
+class TrimmerDecorator < Decorator
+  def correct_name
+    @nameable.correct_name.strip
+  end
+end
```

Comment:
- [OPTIONAL] I would advise that each Ruby class in your project have an individual file because it is industry-standard practice. It's important for Ruby classes and their methods to be in individual files to maintain a clear and organized code structure, making it easier to locate, understand, and modify specific pieces of code, and to encourage the principles of modularity and reusability. 👍 💻